code/entities/cooking/hotdog-roller/extensions/HotdogRoller.State.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace Cinema;
+
+public partial class HotdogRoller
+{
+    public enum State : int
+    {
+        BothOff,
+        BothOn,
+        FrontOn,
+        BackOn
+    }
+    public State MachineState { get; set; }
+    
+    /// <summary>
+    /// Sets initial state for the Hotdog Roller. Default is both rollers off
+    /// </summary>
+    private void SetInitState()
+    {
+        TransitionStateTo(State.BothOff);
+    }
+
+    /// <summary>
+    /// Updates the power state when one of the power switches changes state
+    /// </summary>
+    private void UpdatePowerState()
+    {
+        if (!LeftSwitch.IsOn() && !RightSwitch.IsOn())
```

Comment:
There is a way to simplify this function and have only 3 if statements (1 nested in another)

Using else statements are appropriate in this case too.